odoo/tools/image.py

```diff
@@ -105,7 +107,12 @@ def image_resize_image(base64_source, size=(1024, 1024), encoding='base64', file
         image = image.convert("RGB")
 
     background_stream = io.BytesIO()
-    image.save(background_stream, filetype)
+    # we should only check for 'JPEG' images (PIL can only convert this format into progressive)
+    # and also check that the images are progressive already, to avoid converting a non progressive image
+    if keep_progressive and original.format == 'JPEG' and original.info.get('progression', False) == True:
```

Comment:
dict.get by default returns Falsy value
```suggestion
    if keep_progressive and original.format == 'JPEG' and original.info.get('progression') == True:
```